chase/frontend/components/navbar/settings_sidebar.tsx

```diff
@@ -48,21 +51,22 @@ export default function SettingsSidebar({
       icon: faMoon,
       value: "dark",
     },
-    {
-      name: "System",
-      icon: faDisplay,
-      value: "system",
-    },
     {
       name: "Hoher Kontrast",
       icon: faCircleHalfStroke,
       value: "contrast",
     },
   ];
 
-  const [colortheme, setColortheme] = useState("system");
+  const [colortheme, setColortheme] = useState(
+    localStorage.getItem("theme") || "system",
```

Comment:
@m1212e I am lost. Can you tell me why he thinks localStorage is undefined?

```
- error components/navbar/settings_sidebar.tsx (62:4) @ localStorage
- error ReferenceError: localStorage is not defined
    at SettingsSidebar (./components/navbar/settings_sidebar.tsx:52:89)
  60 | 
  61 |   const [colortheme, setColortheme] = useState(
> 62 |     localStorage.getItem("theme") || "system",
     |    ^
  63 |   );
  64 |   const [isDarkMode, setIsDarkMode] = useState(false);
  65 | 
  ```

Reply:
Try making the component client only, Browser APIs may not be available in SSR

Reply:
This should do the trick

```

  const [colortheme, setColortheme] = useState(
    typeof window !== "undefined"
      ? localStorage.getItem("theme") ?? "system"
      : "system",
  );
  const [isDarkMode, setIsDarkMode] = useState(false);

  const [language, setLanguage] = useState(
    typeof window !== "undefined"
      ? localStorage.getItem("lang") ?? "system"
      : "system",
  );
```

Reply:
Thanks! That seemed so overcomplicated to me, that I thought there must be a more slick solution... I'll implement it ;)